lib/launcher.js

```diff
@@ -439,8 +458,22 @@ class Launcher {
         this.proc.stderr.on('data', handleLog)
     }
 
-    async stop (clean) {
-        info('Stopping Node-RED')
+    async stop (clean, reason) {
+        if (this.installProcess && this.state === States.INSTALLING) {
+            // If the launcher is currently installing, we should try not to interrupt this
+            // to avoid corruption (NPM can leave temporary directories preventing future installs)
+            // We should wait for the install to finish before stopping
+            // give it a few seconds to finish
+            const timeout = new Promise(resolve => setTimeout(resolve, 10000))
+            await Promise.race([this.installProcess, timeout])
+            // now proceed with stopping, regardless of whether the install finished
+        }
+        let finalState = States.STOPPED
+        this.stopReason = reason || 'shutdown'
+        info('Stopping Node-RED. Reason: ' + this.reason)
+        if (reason === States.SUSPENDED) {
+            finalState = States.SUSPENDED
+        }
```

Comment:
```suggestion
        this.stopReason = reason || 'shutdown'
        info('Stopping Node-RED. Reason: ' + this.stopReason)
        if (this.stopReason === States.SUSPENDED) {
            finalState = States.SUSPENDED
        }
```

@hardillb apologies, i messed up.  fixed now.